src/NodaTime/Text/PeriodPattern.cs

```diff
@@ -184,6 +184,14 @@ public StringBuilder AppendFormat(Period value, StringBuilder builder)
             {
                 Preconditions.CheckNotNull(value, nameof(value));
                 Preconditions.CheckNotNull(builder, nameof(builder));
+                // Always ensure we've got *some* unit to ensure the result is valid in ISO-8601; arbitrarily pick days.
+                // Note: "P0S" might be nicer here, but NormalizingIsoPatternImpl picked "P0D" a long time ago.
```

Comment:
"... and we want to be consistent"? (or is there some other reason they need to be the same?)

Reply:
Nope, that makes sense. Will add that, rebase, force push, and merge.